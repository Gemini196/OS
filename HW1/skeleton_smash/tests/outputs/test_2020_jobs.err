/bin/bash: line 0: kill: (3) - No such process
