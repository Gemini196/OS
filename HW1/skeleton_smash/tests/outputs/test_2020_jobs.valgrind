==68762== Memcheck, a memory error detector
==68762== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==68762== Using Valgrind-3.13.0 and LibVEX; rerun with -h for copyright info
==68762== Command: /home/student/Desktop/OS/HW1/skeleton_smash/smash
==68762== Parent PID: 68758
==68762== 
==68762== 
==68762== Process terminating with default action of signal 6 (SIGABRT)
==68762==    at 0x541DE87: raise (raise.c:51)
==68762==    by 0x541F7F0: abort (abort.c:79)
==68762==    by 0x4ECA956: ??? (in /usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.25)
==68762==    by 0x4ED0AE5: ??? (in /usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.25)
==68762==    by 0x4ED0B20: std::terminate() (in /usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.25)
==68762==    by 0x4ED0D53: __cxa_throw (in /usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.25)
==68762==    by 0x4ECC79E: ??? (in /usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.25)
==68762==    by 0x4F652DC: void std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_construct<char const*>(char const*, char const*, std::forward_iterator_tag) (in /usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.25)
==68762==    by 0x10D5C4: BackgroundCommand::execute() (Commands.cpp:560)
==68762==    by 0x10F120: SmallShell::executeCommand(char const*) (Commands.cpp:916)
==68762==    by 0x1109A1: main (smash.cpp:51)
==68762== 
==68762== FILE DESCRIPTORS: 6 open at exit.
==68762== Open file descriptor 1: /home/student/Desktop/OS/HW1/skeleton_smash/tests/outputs/test_2020_jobs.out
==68762==    at 0x54EF957: dup2 (syscall-template.S:78)
==68762==    by 0x1108F8: main (smash.cpp:41)
==68762== 
==68762== Open file descriptor 0:
==68762==    at 0x54EF957: dup2 (syscall-template.S:78)
==68762==    by 0x1108A6: main (smash.cpp:33)
==68762== 
==68762== Open file descriptor 5:
==68762==    at 0x54EF927: dup (syscall-template.S:78)
==68762==    by 0x110A67: __static_initialization_and_destruction_0(int, int) (smash.cpp:9)
==68762==    by 0x110A83: _GLOBAL__sub_I_stdout_dp (smash.cpp:57)
==68762==    by 0x110AFC: __libc_csu_init (in /home/student/Desktop/OS/HW1/skeleton_smash/smash)
==68762==    by 0x5400C17: (below main) (libc-start.c:266)
==68762== 
==68762== Open file descriptor 4: /home/student/Desktop/OS/HW1/skeleton_smash/tests/outputs/test_2020_jobs.out
==68762==    at 0x54EF927: dup (syscall-template.S:78)
==68762==    by 0x110A57: __static_initialization_and_destruction_0(int, int) (smash.cpp:8)
==68762==    by 0x110A83: _GLOBAL__sub_I_stdout_dp (smash.cpp:57)
==68762==    by 0x110AFC: __libc_csu_init (in /home/student/Desktop/OS/HW1/skeleton_smash/smash)
==68762==    by 0x5400C17: (below main) (libc-start.c:266)
==68762== 
==68762== Open file descriptor 3: /home/student/Desktop/OS/HW1/skeleton_smash/tests/outputs/test_2020_jobs.valgrind
==68762==    <inherited from parent>
==68762== 
==68762== Open file descriptor 2: /home/student/Desktop/OS/HW1/skeleton_smash/tests/outputs/test_2020_jobs.err
==68762==    <inherited from parent>
==68762== 
==68762== 
==68762== HEAP SUMMARY:
==68762==     in use at exit: 1,677 bytes in 31 blocks
==68762==   total heap usage: 111 allocs, 80 frees, 88,398 bytes allocated
==68762== 
==68762== 3 bytes in 1 blocks are still reachable in loss record 1 of 17
==68762==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==68762==    by 0x10B299: _parseCommandLine(char const*, char**) (Commands.cpp:47)
==68762==    by 0x10D58E: BackgroundCommand::execute() (Commands.cpp:559)
==68762==    by 0x10F120: SmallShell::executeCommand(char const*) (Commands.cpp:916)
==68762==    by 0x1109A1: main (smash.cpp:51)
==68762== 
==68762== 16 bytes in 1 blocks are still reachable in loss record 2 of 17
==68762==    at 0x4C3217F: operator new(unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==68762==    by 0x110845: main (smash.cpp:24)
==68762== 
==68762== 24 bytes in 1 blocks are still reachable in loss record 3 of 17
==68762==    at 0x4C3217F: operator new(unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==68762==    by 0x10E0BD: JobsList::JobsList() (Commands.cpp:707)
==68762==    by 0x110850: main (smash.cpp:24)
==68762== 
==68762== 24 bytes in 1 blocks are still reachable in loss record 4 of 17
==68762==    at 0x4C3217F: operator new(unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==68762==    by 0x110464: __gnu_cxx::new_allocator<std::_List_node<JobsList::JobEntry*> >::allocate(unsigned long, void const*) (new_allocator.h:111)
==68762==    by 0x1103AA: std::allocator_traits<std::allocator<std::_List_node<JobsList::JobEntry*> > >::allocate(std::allocator<std::_List_node<JobsList::JobEntry*> >&, unsigned long) (alloc_traits.h:436)
==68762==    by 0x110258: std::__cxx11::_List_base<JobsList::JobEntry*, std::allocator<JobsList::JobEntry*> >::_M_get_node() (stl_list.h:383)
==68762==    by 0x110083: std::_List_node<JobsList::JobEntry*>* std::__cxx11::list<JobsList::JobEntry*, std::allocator<JobsList::JobEntry*> >::_M_create_node<JobsList::JobEntry* const&>(JobsList::JobEntry* const&) (stl_list.h:572)
==68762==    by 0x10FE85: void std::__cxx11::list<JobsList::JobEntry*, std::allocator<JobsList::JobEntry*> >::_M_insert<JobsList::JobEntry* const&>(std::_List_iterator<JobsList::JobEntry*>, JobsList::JobEntry* const&) (stl_list.h:1801)
==68762==    by 0x10FAED: std::__cxx11::list<JobsList::JobEntry*, std::allocator<JobsList::JobEntry*> >::push_back(JobsList::JobEntry* const&) (stl_list.h:1118)
==68762==    by 0x10E1C5: JobsList::addJob(char const*, int, long, bool) (Commands.cpp:721)
==68762==    by 0x110569: ctrlZHandler(int) (signals.cpp:37)
==68762==    by 0x541DF0F: ??? (in /lib/x86_64-linux-gnu/libc-2.27.so)
==68762==    by 0x54C3334: waitpid (waitpid.c:30)
==68762==    by 0x10D457: ForegroundCommand::execute() (Commands.cpp:534)
==68762==    by 0x10F120: SmallShell::executeCommand(char const*) (Commands.cpp:916)
==68762==    by 0x1109A1: main (smash.cpp:51)
==68762== 
==68762== 24 bytes in 1 blocks are still reachable in loss record 5 of 17
==68762==    at 0x4C3217F: operator new(unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==68762==    by 0x110464: __gnu_cxx::new_allocator<std::_List_node<JobsList::JobEntry*> >::allocate(unsigned long, void const*) (new_allocator.h:111)
==68762==    by 0x1103AA: std::allocator_traits<std::allocator<std::_List_node<JobsList::JobEntry*> > >::allocate(std::allocator<std::_List_node<JobsList::JobEntry*> >&, unsigned long) (alloc_traits.h:436)
==68762==    by 0x110258: std::__cxx11::_List_base<JobsList::JobEntry*, std::allocator<JobsList::JobEntry*> >::_M_get_node() (stl_list.h:383)
==68762==    by 0x110083: std::_List_node<JobsList::JobEntry*>* std::__cxx11::list<JobsList::JobEntry*, std::allocator<JobsList::JobEntry*> >::_M_create_node<JobsList::JobEntry* const&>(JobsList::JobEntry* const&) (stl_list.h:572)
==68762==    by 0x10FE85: void std::__cxx11::list<JobsList::JobEntry*, std::allocator<JobsList::JobEntry*> >::_M_insert<JobsList::JobEntry* const&>(std::_List_iterator<JobsList::JobEntry*>, JobsList::JobEntry* const&) (stl_list.h:1801)
==68762==    by 0x10FAED: std::__cxx11::list<JobsList::JobEntry*, std::allocator<JobsList::JobEntry*> >::push_back(JobsList::JobEntry* const&) (stl_list.h:1118)
==68762==    by 0x10E1C5: JobsList::addJob(char const*, int, long, bool) (Commands.cpp:721)
==68762==    by 0x110569: ctrlZHandler(int) (signals.cpp:37)
==68762==    by 0x541DF0F: ??? (in /lib/x86_64-linux-gnu/libc-2.27.so)
==68762==    by 0x54C3334: waitpid (waitpid.c:30)
==68762==    by 0x10BA8A: ExternalCommand::execute() (Commands.cpp:162)
==68762==    by 0x10F120: SmallShell::executeCommand(char const*) (Commands.cpp:916)
==68762==    by 0x1109A1: main (smash.cpp:51)
==68762== 
==68762== 24 bytes in 1 blocks are still reachable in loss record 6 of 17
==68762==    at 0x4C3217F: operator new(unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==68762==    by 0x10EF25: SmallShell::CreateCommand(char const*) (Commands.cpp:895)
==68762==    by 0x10F07A: SmallShell::executeCommand(char const*) (Commands.cpp:904)
==68762==    by 0x1109A1: main (smash.cpp:51)
==68762== 
==68762== 24 bytes in 1 blocks are indirectly lost in loss record 7 of 17
==68762==    at 0x4C3217F: operator new(unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==68762==    by 0x10E0BD: JobsList::JobsList() (Commands.cpp:707)
==68762==    by 0x10EB3C: SmallShell::SmallShell() (Commands.cpp:860)
==68762==    by 0x110752: SmallShell::getInstance() (Commands.h:233)
==68762==    by 0x110824: main (smash.cpp:21)
==68762== 
==68762== 40 (16 direct, 24 indirect) bytes in 1 blocks are definitely lost in loss record 8 of 17
==68762==    at 0x4C3217F: operator new(unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==68762==    by 0x10EB31: SmallShell::SmallShell() (Commands.cpp:860)
==68762==    by 0x110752: SmallShell::getInstance() (Commands.h:233)
==68762==    by 0x110824: main (smash.cpp:21)
==68762== 
==68762== 56 bytes in 1 blocks are still reachable in loss record 9 of 17
==68762==    at 0x4C3217F: operator new(unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==68762==    by 0x10E171: JobsList::addJob(char const*, int, long, bool) (Commands.cpp:719)
==68762==    by 0x110569: ctrlZHandler(int) (signals.cpp:37)
==68762==    by 0x541DF0F: ??? (in /lib/x86_64-linux-gnu/libc-2.27.so)
==68762==    by 0x54C3334: waitpid (waitpid.c:30)
==68762==    by 0x10D457: ForegroundCommand::execute() (Commands.cpp:534)
==68762==    by 0x10F120: SmallShell::executeCommand(char const*) (Commands.cpp:916)
==68762==    by 0x1109A1: main (smash.cpp:51)
==68762== 
==68762== 56 bytes in 1 blocks are still reachable in loss record 10 of 17
==68762==    at 0x4C3217F: operator new(unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==68762==    by 0x10E171: JobsList::addJob(char const*, int, long, bool) (Commands.cpp:719)
==68762==    by 0x110569: ctrlZHandler(int) (signals.cpp:37)
==68762==    by 0x541DF0F: ??? (in /lib/x86_64-linux-gnu/libc-2.27.so)
==68762==    by 0x54C3334: waitpid (waitpid.c:30)
==68762==    by 0x10BA8A: ExternalCommand::execute() (Commands.cpp:162)
==68762==    by 0x10F120: SmallShell::executeCommand(char const*) (Commands.cpp:916)
==68762==    by 0x1109A1: main (smash.cpp:51)
==68762== 
==68762== 66 bytes in 1 blocks are still reachable in loss record 11 of 17
==68762==    at 0x4C3217F: operator new(unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==68762==    by 0x4F11C28: std::string::_Rep::_S_create(unsigned long, unsigned long, std::allocator<char> const&) (in /usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.25)
==68762==    by 0x4F11D46: ??? (in /usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.25)
==68762==    by 0x4F13B4F: std::basic_string<char, std::char_traits<char>, std::allocator<char> >::basic_string(char const*, std::allocator<char> const&) (in /usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.25)
==68762==    by 0x4EF5B25: std::logic_error::logic_error(char const*) (in /usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.25)
==68762==    by 0x4EF90C8: std::__throw_logic_error(char const*) (in /usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.25)
==68762==    by 0x4F652DC: void std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_construct<char const*>(char const*, char const*, std::forward_iterator_tag) (in /usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.25)
==68762==    by 0x10D5C4: BackgroundCommand::execute() (Commands.cpp:560)
==68762==    by 0x10F120: SmallShell::executeCommand(char const*) (Commands.cpp:916)
==68762==    by 0x1109A1: main (smash.cpp:51)
==68762== 
==68762== 144 bytes in 1 blocks are possibly lost in loss record 12 of 17
==68762==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==68762==    by 0x4ECF8FF: __cxa_allocate_exception (in /usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.25)
==68762==    by 0x4EF90B2: std::__throw_logic_error(char const*) (in /usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.25)
==68762==    by 0x4F652DC: void std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_construct<char const*>(char const*, char const*, std::forward_iterator_tag) (in /usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.25)
==68762==    by 0x10D5C4: BackgroundCommand::execute() (Commands.cpp:560)
==68762==    by 0x10F120: SmallShell::executeCommand(char const*) (Commands.cpp:916)
==68762==    by 0x1109A1: main (smash.cpp:51)
==68762== 
==68762== 160 bytes in 1 blocks are still reachable in loss record 13 of 17
==68762==    at 0x4C3289F: operator new[](unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==68762==    by 0x10D543: BackgroundCommand::execute() (Commands.cpp:553)
==68762==    by 0x10F120: SmallShell::executeCommand(char const*) (Commands.cpp:916)
==68762==    by 0x1109A1: main (smash.cpp:51)
==68762== 
==68762== 192 bytes in 8 blocks are still reachable in loss record 14 of 17
==68762==    at 0x4C3217F: operator new(unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==68762==    by 0x110464: __gnu_cxx::new_allocator<std::_List_node<JobsList::JobEntry*> >::allocate(unsigned long, void const*) (new_allocator.h:111)
==68762==    by 0x1103AA: std::allocator_traits<std::allocator<std::_List_node<JobsList::JobEntry*> > >::allocate(std::allocator<std::_List_node<JobsList::JobEntry*> >&, unsigned long) (alloc_traits.h:436)
==68762==    by 0x110258: std::__cxx11::_List_base<JobsList::JobEntry*, std::allocator<JobsList::JobEntry*> >::_M_get_node() (stl_list.h:383)
==68762==    by 0x110083: std::_List_node<JobsList::JobEntry*>* std::__cxx11::list<JobsList::JobEntry*, std::allocator<JobsList::JobEntry*> >::_M_create_node<JobsList::JobEntry* const&>(JobsList::JobEntry* const&) (stl_list.h:572)
==68762==    by 0x10FE85: void std::__cxx11::list<JobsList::JobEntry*, std::allocator<JobsList::JobEntry*> >::_M_insert<JobsList::JobEntry* const&>(std::_List_iterator<JobsList::JobEntry*>, JobsList::JobEntry* const&) (stl_list.h:1801)
==68762==    by 0x10FAED: std::__cxx11::list<JobsList::JobEntry*, std::allocator<JobsList::JobEntry*> >::push_back(JobsList::JobEntry* const&) (stl_list.h:1118)
==68762==    by 0x10E1C5: JobsList::addJob(char const*, int, long, bool) (Commands.cpp:721)
==68762==    by 0x10BA40: ExternalCommand::execute() (Commands.cpp:156)
==68762==    by 0x10F120: SmallShell::executeCommand(char const*) (Commands.cpp:916)
==68762==    by 0x1109A1: main (smash.cpp:51)
==68762== 
==68762== 200 bytes in 1 blocks are still reachable in loss record 15 of 17
==68762==    at 0x4C3289F: operator new[](unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==68762==    by 0x10EB4E: SmallShell::SmallShell() (Commands.cpp:861)
==68762==    by 0x110752: SmallShell::getInstance() (Commands.h:233)
==68762==    by 0x110824: main (smash.cpp:21)
==68762== 
==68762== 200 bytes in 1 blocks are still reachable in loss record 16 of 17
==68762==    at 0x4C3289F: operator new[](unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==68762==    by 0x10D551: BackgroundCommand::execute() (Commands.cpp:554)
==68762==    by 0x10F120: SmallShell::executeCommand(char const*) (Commands.cpp:916)
==68762==    by 0x1109A1: main (smash.cpp:51)
==68762== 
==68762== 448 bytes in 8 blocks are still reachable in loss record 17 of 17
==68762==    at 0x4C3217F: operator new(unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==68762==    by 0x10E171: JobsList::addJob(char const*, int, long, bool) (Commands.cpp:719)
==68762==    by 0x10BA40: ExternalCommand::execute() (Commands.cpp:156)
==68762==    by 0x10F120: SmallShell::executeCommand(char const*) (Commands.cpp:916)
==68762==    by 0x1109A1: main (smash.cpp:51)
==68762== 
==68762== LEAK SUMMARY:
==68762==    definitely lost: 16 bytes in 1 blocks
==68762==    indirectly lost: 24 bytes in 1 blocks
==68762==      possibly lost: 144 bytes in 1 blocks
==68762==    still reachable: 1,493 bytes in 28 blocks
==68762==                       of which reachable via heuristic:
==68762==                         stdstring          : 66 bytes in 1 blocks
==68762==         suppressed: 0 bytes in 0 blocks
==68762== 
==68762== For counts of detected and suppressed errors, rerun with: -v
==68762== ERROR SUMMARY: 2 errors from 2 contexts (suppressed: 0 from 0)
