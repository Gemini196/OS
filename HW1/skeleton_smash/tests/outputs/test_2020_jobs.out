smash> smash> smash> smash> smash> smash> smash> smash> [1] sleep 100& : 46739 0 secs
[2] sleep 200& : 46740 0 secs
[3] sleep 300& : 46741 0 secs
[4] sleep 400& : 46742 0 secs
[5] sleep 500& : 46743 0 secs
[6] sleep 600& : 46744 0 secs
[7] sleep 700& : 46745 0 secs
smash> smash> smash> [1] sleep 100& : 46739 0 secs
[2] sleep 200& : 46740 0 secs
[3] sleep 300& : 46741 0 secs
[4] sleep 400& : 46742 0 secs
[5] sleep 500& : 46743 0 secs
[6] sleep 600& : 46744 0 secs
[7] sleep 700& : 46745 0 secs
[8] sleep 800& : 46746 0 secs
[9] sleep 900& : 46747 0 secs
smash> signal number 9 was sent to pid 46741
smash> [1] sleep 100& : 46739 1 secs
[2] sleep 200& : 46740 1 secs
[4] sleep 400& : 46742 1 secs
[5] sleep 500& : 46743 1 secs
[6] sleep 600& : 46744 1 secs
[7] sleep 700& : 46745 1 secs
[8] sleep 800& : 46746 1 secs
[9] sleep 900& : 46747 1 secs
smash> smash> [1] sleep 100& : 46739 1 secs
[2] sleep 200& : 46740 1 secs
[4] sleep 400& : 46742 1 secs
[5] sleep 500& : 46743 1 secs
[6] sleep 600& : 46744 1 secs
[7] sleep 700& : 46745 1 secs
[8] sleep 800& : 46746 1 secs
[9] sleep 900& : 46747 1 secs
[10] sleep 1000& : 46942 0 secs
smash> signal number 9 was sent to pid 46942
smash> [1] sleep 100& : 46739 2 secs
[2] sleep 200& : 46740 2 secs
[4] sleep 400& : 46742 2 secs
[5] sleep 500& : 46743 2 secs
[6] sleep 600& : 46744 2 secs
[7] sleep 700& : 46745 2 secs
[8] sleep 800& : 46746 2 secs
[9] sleep 900& : 46747 2 secs
smash> smash> [1] sleep 100& : 46739 2 secs
[2] sleep 200& : 46740 2 secs
[4] sleep 400& : 46742 2 secs
[5] sleep 500& : 46743 2 secs
[6] sleep 600& : 46744 2 secs
[7] sleep 700& : 46745 2 secs
[8] sleep 800& : 46746 2 secs
[9] sleep 900& : 46747 2 secs
[10] sleep 1001& : 47027 0 secs
smash> sleep 1001& : 47027
smash: got ctrl-Z
smash: process 47027 was stopped
smash> [1] sleep 100& : 46739 3 secs
[2] sleep 200& : 46740 3 secs
[4] sleep 400& : 46742 3 secs
[5] sleep 500& : 46743 3 secs
[6] sleep 600& : 46744 3 secs
[7] sleep 700& : 46745 3 secs
[8] sleep 800& : 46746 3 secs
[9] sleep 900& : 46747 3 secs
[10] sleep 1001& : 47027 0 secs (stopped)
smash> sleep 1001& : 47027
smash: got ctrl-Z
smash: process 47027 was stopped
smash> [1] sleep 100& : 46739 3 secs
[2] sleep 200& : 46740 3 secs
[4] sleep 400& : 46742 3 secs
[5] sleep 500& : 46743 3 secs
[6] sleep 600& : 46744 3 secs
[7] sleep 700& : 46745 3 secs
[8] sleep 800& : 46746 3 secs
[9] sleep 900& : 46747 3 secs
[10] sleep 1001& : 47027 0 secs (stopped)
smash> sleep 1001& : 47027
smash: got ctrl-Z
smash: process 47027 was stopped
smash> [1] sleep 100& : 46739 4 secs
[2] sleep 200& : 46740 4 secs
[4] sleep 400& : 46742 4 secs
[5] sleep 500& : 46743 4 secs
[6] sleep 600& : 46744 4 secs
[7] sleep 700& : 46745 4 secs
[8] sleep 800& : 46746 4 secs
[9] sleep 900& : 46747 4 secs
[10] sleep 1001& : 47027 0 secs (stopped)
smash> sleep 1001& : 47027
smash: got ctrl-Z
smash: process 47027 was stopped
smash> [1] sleep 100& : 46739 4 secs
[2] sleep 200& : 46740 4 secs
[4] sleep 400& : 46742 4 secs
[5] sleep 500& : 46743 4 secs
[6] sleep 600& : 46744 4 secs
[7] sleep 700& : 46745 4 secs
[8] sleep 800& : 46746 4 secs
[9] sleep 900& : 46747 4 secs
[10] sleep 1001& : 47027 0 secs (stopped)
smash> smash: got ctrl-C
smash: process 0 was killed
smash: got ctrl-C
[1] sleep 100& : 46739 5 secs
[2] sleep 200& : 46740 5 secs
[4] sleep 400& : 46742 5 secs
[5] sleep 500& : 46743 5 secs
[6] sleep 600& : 46744 5 secs
[7] sleep 700& : 46745 5 secs
[8] sleep 800& : 46746 5 secs
[9] sleep 900& : 46747 5 secs
[10] sleep 1001& : 47027 1 secs (stopped)
smash> smash: got ctrl-C
smash: process 0 was killed
smash: got ctrl-C
smash: got ctrl-C
smash: got ctrl-C
smash: process 47176 was killed
smash> 